### PR TITLE
Add python3 support

### PR DESCRIPTION
**Pull request by toopy**

### CODE REVIEW

**Comment by toopy** [comment]
@k4nar OPM ?

File: populous/compat.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import sys
+
+from past.builtins import xrange
```
Comment (k4nar) [inline populous/compat.py]: I think I prefer the "forward-compatible" approach, so `from builtins import range`.

This means we need to import it everywhere `range` is used, and convert `range(...)` to `list(range(...))` where needed.

Comment (toopy) [reply populous/compat.py]: ok

File: populous/compat.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import sys
 
-from past.builtins import xrange
+from builtins import range
```
Comment (k4nar) [inline populous/compat.py]: `from future.builtins import range` is a bit more explicit

File: populous/bloom.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from peloton_bloomfilters import BloomFilter as _BloomFilter
+from pybloom_live import BloomFilter as _BloomFilter
```
Comment (k4nar) [inline populous/bloom.py]: Actually this lib is very slow. The better lib I've found for Python 3 is `pybloomfilter3`.

The best solution would be to use either `peloton_bloomfilters` or `pybloomfilter3` according to which Python version we are using.

File: populous/compat.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+from past.builtins import basestring
+
+PY2 = sys.version_info[0] == 2
+
+
+try:
+    from itertools import izip
+except ImportError:
+    izip = zip
```
Comment (k4nar) [inline populous/compat.py]: same remark than for `range` I prefer `zip = izip`. But I'm pretty sure futures already has that.

File: populous/compat.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+try:
+    from string import letters
+except ImportError:
+    from string import ascii_letters as letters
```
Comment (k4nar) [inline populous/compat.py]: same

File: populous/exceptions.py [hunk 6]
```diff
@@ -1,5 +1,8 @@
 class PopulousError(Exception):
-    pass
+
+    @property
+    def message(self):
+        return self.args[0]
```
Comment (k4nar) [inline populous/exceptions.py]: actually I think we should overload `__str__` in `ValidationError` instead.

File: tests/test_generation.py [hunk 7]
```diff
@@ -123,7 +125,7 @@ def select(self, table, *args, **kwargs):
 
 def test_blueprint_generate(mocker):
     import random
-    random.seed(42)
+    random.seed(24)
```
Comment (k4nar) [inline tests/test_generation.py]: just curious, why changing the seed ? :)

File: populous/bloom.py (hunk 3 above)
Comment (toopy) [reply populous/bloom.py]: I can't find pybloomfilter3 on pypi ? How should we install it ?

Comment (k4nar) [reply populous/bloom.py]: Sorry actually it's `pybloomfiltermmap3`.

File: tests/test_generation.py (hunk 7 above)
Comment (toopy) [reply tests/test_generation.py]: Nothing important :)

File: populous/bloom.py (hunk 3 above)
Comment (toopy) [reply populous/bloom.py]: We know have a warning due to the following line :/

- https://github.com/prashnts/pybloomfiltermmap3/blob/master/src/pybloomfilter.pyx#L116

```bash
tests/test_generators.py::test_yaml
  /home/florent/work/peoplecore-sandbox/var/populous/populous/bloom.py:11: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    self._filters.append(self._create_filter())
```

File: tests/test_bloom.py [hunk 8]
```diff
@@ -20,28 +21,28 @@ def test_bloom_filter():
...
         assert x in bf
 
     errors = 0
-    for x in xrange(1000, 11000):
+    for x in range(1000, 11000):
         if x in bf:
             errors += 1
 
-    assert errors / 10000. < 0.01
+    assert errors / 10000. < 0.0122
```
Comment (k4nar) [inline tests/test_bloom.py]: do we still need that change ?

**Review by k4nar — APPROVED** [review]
👍

**Comment by toopy** [comment]
@k4nar I just cleaned the branch. If it's good for you, I let you merge ;)

**Pull request merged**